lib/remap-functions/src/to_unix_timestamp.rs

```diff
@@ -0,0 +1,194 @@
+use remap::prelude::*;
+use std::str::FromStr;
+
+#[derive(Clone, Copy, Debug)]
+pub struct ToUnixTimestamp;
+
+impl Function for ToUnixTimestamp {
+    fn identifier(&self) -> &'static str {
+        "to_unix_timestamp"
+    }
+
+    fn parameters(&self) -> &'static [Parameter] {
+        &[
+            Parameter {
+                keyword: "value",
+                accepts: |v| matches!(v, Value::Timestamp(_)),
+                required: true,
+            },
+            Parameter {
+                keyword: "unit",
+                accepts: |v| matches!(v, Value::Bytes(_)),
+                required: false,
+            },
+        ]
+    }
+
+    fn compile(&self, mut arguments: ArgumentList) -> Result<Box<dyn Expression>> {
+        let value = arguments.required("value")?.boxed();
+
+        let unit = arguments
+            .optional_enum("unit", &Unit::all_str())?
+            .map(|s| Unit::from_str(&s).expect("validated enum"))
+            .unwrap_or_default();
+
+        Ok(Box::new(ToUnixTimestampFn { value, unit }))
+    }
+}
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+enum Unit {
+    Seconds,
+    Milliseconds,
+    Nanoseconds,
+}
+
+impl Unit {
+    fn all_str() -> Vec<&'static str> {
+        use Unit::*;
+
+        vec![Seconds, Milliseconds, Nanoseconds]
+            .into_iter()
+            .map(|u| u.as_str())
+            .collect::<Vec<_>>()
+    }
+
+    const fn as_str(self) -> &'static str {
+        use Unit::*;
+
+        match self {
+            Seconds => "seconds",
+            Milliseconds => "milliseconds",
+            Nanoseconds => "nanoseconds",
+        }
+    }
+}
+
+impl Default for Unit {
```

Comment:
Nit: just noting we use the `derivative` crate for defaulting enums in some places: https://mcarton.github.io/rust-derivative/latest/index.html

Reply:
Good to know! I'm going to forgo that here, as there's another function that does this "the hard way" and that would be out of scope. But I may follow up.